volttron/platform/vip/agent/subsystems/auth.py

```diff
@@ -76,30 +79,78 @@ def onsetup(sender, **kwargs):
+    def connect_remote_platform(self, address, server_key=None, remote_bus_type=None, agent_class=None):
         from volttron.platform.vip.agent.utils import build_agent
         from volttron.platform.web import DiscoveryInfo
+        from volttron.platform.vip.agent import Agent
+        from volttron.platform.web import DiscoveryError
+
+        if agent_class is None:
+            agent_class = Agent
+
+        parsed_address = urlparse.urlparse(address)
+
+        value = None
+        if parsed_address.scheme in ('https', 'http'):
+            try:
+                # We need to discover which type of bus is at the other end.
+                info = DiscoveryInfo.request_discovery_info(address)
+
+                remote_identity = "{}.{}.{}".format(info.instance_name,
+                                                    get_platform_instance_name(),
+                                                    self._core().identity)
+
+                if info.messagebus_type == 'rmq' and get_messagebus() == 'zmq':
+                    ValueError("ZMQ -> RMQ connection not supported at this time.")
+
+                # If this is present we can assume we are going to connect to rabbit
+                if info.messagebus_type == 'rmq':
+
+                    # Discovery info for external platform
+                    value = self.request_cert(address)
+
+                    _log.debug("RESPONSE VALUE WAS: {}".format(value))
+                    if not isinstance(value, tuple):
+                        info = DiscoveryInfo.request_discovery_info(address)
+                        remote_rmq_user = remote_identity
+                        remote_rmq_address = self._core().rmq_mgmt.build_remote_connection_param(
+                            remote_rmq_user,
+                            info.vc_rmq_address)
+
+                        # remote_identity = "{}.{}".format(get_platform_instance_name(), self.core.identity)
+                        return build_agent(identity=remote_rmq_user,
+                                           address=remote_rmq_address,
+                                           instance_name=info.instance_name)
+
+                else:
+                    # TODO: cache the connection so we don't always have to ping
+                    #       the server to connect.
+
+                    # This branch happens when the message bus is not the same note
+                    # this writes to the agent-data directory of this agent if the agent
+                    # is installed.
+                    if get_messagebus() == 'rmq':
+                        if not os.path.exists("keystore.json"):
+                            with open("keystore.json", 'w') as fp:
+                                fp.write(json.dumps(KeyStore.generate_keypair_dict()))
+
+                        with open("keystore.json") as fp:
+                            keypair = json.loads(fp.read())
+
+                    return build_agent(agent_class=agent_class,
```

Comment:
Could set it as the variable 'value' so that there is a single return statement at the end.